qt/scientific_interfaces/Muon/MuonAnalysisResultTableCreator.cpp

```diff
@@ -129,9 +139,13 @@ ITableWorkspace_sptr MuonAnalysisResultTableCreator::createTable() const {
   for (const auto &log : m_logs) {
 
     auto val = valMap[log];
-    // multiple files use strings due to x-y format
-    if (val.canConvert<double>() && !log.endsWith(" (text)")) {
+    auto dashIndex = val.toString().indexOf("-");
 
+    // multiple files use strings due to x-y format
+    if (dashIndex != 0 && dashIndex != -1) {
+      addColumnToTable(table, "str", log.toStdString(), PLOT_TYPE_X);
```

Comment:
This is the same expression as the else option below, can the two be combined in some way? As a more general point is there any reason we can't get the values of everything as strings and add them to the table this way using `addColumnToTable` for all cases? This point also applies to some of the cases below.

Reply:
Unfortunately they want numeric columns when possible, so they can be easily plotted. We could remove the else by inverting the following if statements and making them part of this one.

Reply:
Looking at it I think trying to condense will probably lead to more bugs